deepchem/transformers/__init__.py

```diff
@@ -172,22 +172,60 @@ def untransform(self, z):
 
 class LogTransformer(Transformer):
 
+  def __init__(self, transform_X=False, transform_y=False,
+               features=None, tasks=None,
+               dataset=None):
+    self.features=features
+    self.tasks=tasks
+    """Initialize log  transformation."""
+    super(LogTransformer, self).__init__(
+        transform_X=transform_X, transform_y=transform_y,
+        dataset=dataset)
+
   def transform_row(self, i, df, data_dir):
     """Logarithmically transforms data in dataset."""
+    """Select features and tasks of interest for transformation."""
     row = df.iloc[i]
     if self.transform_X:
       X = load_from_disk(os.path.join(data_dir, row['X-transformed']))
-      X = np.log(X+1)
+      end_feat=len(X[1,:])
```

Comment:
Couple questions/comments:
- If I understand correctly, `end_feat` is intended to be the number of features for datapoints in `X` correct? Should rename to `num_features` and probably define as `num_features = len(X[0])` to make it more clear what's happening.
- We should handle the edge case where `self.features == None`. In that case, we should transform everything, since that's what all the existing invocations of `LogTransformer` expect.